### PR TITLE
feat: add 3rd batch of TODO removals

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: [WEB-905](https://determinedai.atlassian.net/browse/WEB-905)

We have quite a few TODOS in our codebase, so, this PR intends to resolve a few. The idea is to submit batches of PR's to help merging them faster.

The topic for this batch is around the usage of `Loadable.match` in a few more places and general `TODO's`.

## Test Plan

The overall usage shouldn't have changed, only a few fallbacks with `<Spinner />` were added, still, here are the pages that should be checked (a quick refresh or just arriving at the page should do).

- Cluster and Clusters
- Experiment Details
- ResourcepoolDetails
- TrialDetails (around charts also)

"non-page" changes

- Check places where we use the `UPlotChart` and `LineChart`
- useModalCreateGroup
- useModalCreateUser

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-905]: https://determinedai.atlassian.net/browse/WEB-905?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gt2345 — COMMENTED** [review]
Overall looks good, but some pages end up having 2 spinners and I'm not sure that's ideal.
Screen Shot 2023-03-01 at 11 29 00 AM
Screen Shot 2023-03-01 at 11 26 29 AM

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx [hunk 1]
```diff
@@ -121,31 +126,31 @@ const ModalForm: React.FC<Props> = ({ form, users, group, groupRoles }) => {
...
+                {roles // This is ok as the Spinner has conditionalRender active
+                  ? roles.map((r) => (
                     <Select.Option
                       disabled={groupRoles?.find((gr) => gr.id === r.id)?.fromWorkspace?.length}
                       key={r.id}
                       value={r.id}>
                       {r.name}
                     </Select.Option>
-                  )),
-                NotLoaded: () => undefined, // TODO show spinner when this is loading
```
Comment (ClaireNeveu) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx:141]: This should be blocked on Select having a loading state. We don't want to replace the field with a spinner.

File: webui/react/src/pages/ResourcepoolDetail.tsx [hunk 2]
```diff
@@ -50,11 +50,13 @@ export const DEFAULT_POOL_TAB_KEY = TabType.Active;
 
 const ResourcepoolDetailInner: React.FC = () => {
   const { poolname, tab } = useParams<Params>();
-  const resourcePools = Loadable.getOrElse([], useObservable(useClusterStore().resourcePools)); // TODO show spinner when this is loading
```
Comment (ClaireNeveu) [inline webui/react/src/pages/ResourcepoolDetail.tsx:53]: You don't actually seem to be resolving this TODO.

File: webui/react/src/pages/Cluster/ClusterOverallBar.tsx [hunk 3]
```diff
@@ -3,65 +3,79 @@ import React, { useMemo } from 'react';
...
+import { useClusterStore } from 'stores/cluster';
 import { ShirtSize } from 'themes';
 import { ResourceType } from 'types';
 import { getSlotContainerStates } from 'utils/cluster';
 import { Loadable } from 'utils/loadable';
 import { useObservable } from 'utils/observable';
 
 export const ClusterOverallBar: React.FC = () => {
-  const overview = Loadable.getOrElse(
-    initClusterOverview,
```
Comment (ClaireNeveu) [inline webui/react/src/pages/Cluster/ClusterOverallBar.tsx:15]: Can we remove `initClusterOverview` now?

**Review by gt2345 — APPROVED** [review]
LGTM

**Pull request merged**